### PR TITLE
New mob fire!

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

This is a port of TauCeti's pending fire icons, I really like them, and have been given spritetainer approval to port these.

This PR also changes how fire is handled, or at least, returns the way it used to be handled, while also updating it to use species to choose what kind of fire overlay to use. This variable was set before... but never used...

... This resulted in monkeys not having the correct fire overlay! Unbelivable. That's been fixed however!

Here's a preview:

https://user-images.githubusercontent.com/9026500/166853827-b5512f4d-cea7-4421-bd86-3637092ec611.mp4

Small fire:
image

Big fire:
image

Monke!
image

Original PR: https://github.com/TauCetiStation/TauCetiClassic/pull/9238

Creator: DarthSidiousPalpatine
Credits go to them for these pretty awesome sprites.

## Why It's Good For The Game

Well, I think these look good, it also fixes some strange behavior with the update fire system.

## Changelog
:cl:
imageadd: Mob fire sprites have been updated!
refactor: The way mobs update fire has been refactored. If you spot any anomalies, please report them.
fix: Monkeys will now actually show the correct fire sprite!
/:cl:

### CODE REVIEW

**Comment by dawsonkeyes** [comment]
YES I'M ON FIRE

**Review by Imaginos16 — APPROVED** [review]
hot

File: code/modules/mob/living/carbon/carbon_update_icons.dm [hunk 1]
```diff
@@ -79,14 +79,24 @@
...
 	remove_overlay(FIRE_LAYER)
+	remove_overlay(SECONDARY_FIRE_LAYER)
+
 	if(on_fire || HAS_TRAIT(src, TRAIT_PERMANENTLY_ONFIRE))
-		var/mutable_appearance/new_fire_overlay = mutable_appearance('icons/mob/OnFire.dmi', fire_icon, -FIRE_LAYER)
-		new_fire_overlay.appearance_flags = RESET_COLOR
-		overlays_standing[FIRE_LAYER] = new_fire_overlay
+		var/fire_override = dna?.species?.fire_overlay
+		if(!fire_override)
+			fire_override = "human"
```
Comment (Kapu1178) [inline code/modules/mob/living/carbon/carbon_update_icons.dm:89]:
```suggestion
		fire_override |= "human"
```

**Comment by Gandalf2k15** [comment]
I'm aware the layer is incorrect, I will fix it shortly.

File: code/modules/mob/living/carbon/carbon_update_icons.dm [hunk 2]
```diff
@@ -79,14 +79,24 @@
...
+		var/mutable_appearance/small_fire_overlay = mutable_appearance('icons/mob/onfire.dmi', "[fire_override]_small_fire", -FIRE_LAYER)
+		small_fire_overlay.appearance_flags = RESET_COLOR
+		overlays_standing[FIRE_LAYER] = small_fire_overlay
```
Comment (Kapu1178) [inline code/modules/mob/living/carbon/carbon_update_icons.dm:92]:
```suggestion
		overlays_standing[FIRE_LAYER] = mutable_appearance('icons/mob/onfire.dmi', "[fire_override]_small_fire", -FIRE_LAYER, appearance_flags = RESET_COLOR)
```

File: code/modules/mob/living/carbon/carbon_update_icons.dm [hunk 3]
```diff
@@ -79,14 +79,24 @@
...
+		if(fire_stacks > MOB_BIG_FIRE_STACK_THRESHOLD)
+			var/mutable_appearance/big_fire_overlay = mutable_appearance('icons/mob/onfire.dmi', "[fire_override]_big_fire", -SECONDARY_FIRE_LAYER)
+			big_fire_overlay.appearance_flags = RESET_COLOR
+			overlays_standing[SECONDARY_FIRE_LAYER] = big_fire_overlay
```
Comment (Kapu1178) [inline code/modules/mob/living/carbon/carbon_update_icons.dm:96]:
```suggestion
			overlays_standing[SECONDARY_FIRE_LAYER] = mutable_appearance('icons/mob/onfire.dmi', "[fire_override]_big_fire", -SECONDARY_FIRE_LAYER, appearance_flags = RESET_COLOR)
```

File: code/modules/mob/living/carbon/carbon_update_icons.dm [hunk 4]
```diff
@@ -79,14 +79,24 @@
...
-/mob/living/carbon/update_fire(fire_icon = "Generic_mob_burning")
+/mob/living/carbon/update_fire()
 	remove_overlay(FIRE_LAYER)
+	remove_overlay(SECONDARY_FIRE_LAYER)
+
 	if(on_fire || HAS_TRAIT(src, TRAIT_PERMANENTLY_ONFIRE))
-		var/mutable_appearance/new_fire_overlay = mutable_appearance('icons/mob/OnFire.dmi', fire_icon, -FIRE_LAYER)
-		new_fire_overlay.appearance_flags = RESET_COLOR
-		overlays_standing[FIRE_LAYER] = new_fire_overlay
+		var/fire_override = dna?.species?.fire_overlay
```
Comment (Kapu1178) [inline code/modules/mob/living/carbon/carbon_update_icons.dm:87]:
```suggestion
		var/fire_override = dna?.species.fire_overlay
```
DNA should never be null here but sanity is fine here. Species will NEVER be null if dna is present I don't believe.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

**Comment by SmArtKar** [comment]
Conflicts with datumized fire over here https://github.com/tgstation/tgstation/pull/66573

**Comment by SmArtKar** [comment]
Conflicts with datumized fire over here https://github.com/tgstation/tgstation/pull/66573

**Comment by Fikou** [comment]
minecraft fire

**Comment by UDaV73rus** [comment]
> Creator: DarthSidiousPalpatine
> Credits go to them for these pretty awesome sprites.

@DarthSidiousPalpatine - code for TauCeti
@UDaV73rus (oh, its me!) - sprites

I can sprite for you gradations of fire, if you need it.
Will include in next bunch of fire sprites

File: code/modules/mob/living/carbon/carbon_update_icons.dm [hunk 5]
```diff
@@ -79,14 +79,19 @@
...
+/mob/living/carbon/update_fire()
 	remove_overlay(FIRE_LAYER)
+	remove_overlay(SECONDARY_FIRE_LAYER)
+
 	if(on_fire || HAS_TRAIT(src, TRAIT_PERMANENTLY_ONFIRE))
-		var/mutable_appearance/new_fire_overlay = mutable_appearance('icons/mob/OnFire.dmi', fire_icon, -FIRE_LAYER)
-		new_fire_overlay.appearance_flags = RESET_COLOR
-		overlays_standing[FIRE_LAYER] = new_fire_overlay
+		var/fire_override = dna?.species.fire_overlay
+		fire_override ||= "human"
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/carbon_update_icons.dm:88]:
```suggestion
		var/fire_override = dna?.species.fire_overlay || "human"
```

File: code/__DEFINES/mobs.dm [hunk 6]
```diff
@@ -422,7 +422,7 @@
 #define MAX_REVIVE_FIRE_DAMAGE 180
 #define MAX_REVIVE_BRUTE_DAMAGE 180
 
-#define HUMAN_FIRE_STACK_ICON_NUM 3
+#define MOB_BIG_FIRE_STACK_THRESHOLD 3
```
Comment (Mothblocks) [inline code/__DEFINES/mobs.dm:425]: Document this

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Nit

**Comment by dragomagol** [comment]
Does this fix #12605 ?

**Comment by Farquaar** [comment]
I hope the big fire keeps that "totally engulfed in flames" look in-game, but it looks great in the IconDiffBot

**Pull request closed without merging**